src/interfaces/traits.rs

```diff
@@ -0,0 +1,12 @@
+use crate::errors::TockloaderError;
+use enum_dispatch::enum_dispatch;
+
+#[enum_dispatch]
+pub trait BoardInterface {
+    fn open(&mut self) -> Result<(), TockloaderError>;
+}
+
+#[enum_dispatch]
+pub trait BytesReader {
```

Comment:
"Reader" seems too specific to me. The three basic operations every channel has to support are read, write, and erase. I think it would make sense to group those three functions in one trait.

Then, all of the "other" operations (namely `open()`, but could also be recognize_board and others) could be grouped into a trait as you have done.

Reply:
`BytesReader` was just supposed to be a placeholder that would exemplify `enum_dispatch`'s ability to work with multiple traits at once. I was planning on creating several traits (and their corresponding implementations marked with `todo!()` alongisde ) in the next pull request, after this one is more or less resolved. 

I'm probably going to delete this trait, and opt for comments to explain the crate's functionality